src/BenchmarkDotNet/Extensions/Hashing.cs

```diff
@@ -0,0 +1,117 @@
+// file copied from https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.Core/Utilities/Hashing.cs
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace BenchmarkDotNet.Extensions
+{
+    internal static class Hashing
+    {
+        private const uint _defaultSeed = (5381 << 16) + 5381;
+
+        internal static uint HashString(ReadOnlySpan<char> str) => MurmurHash(_defaultSeed, str);
+
+        /// <summary>
+        /// Implements the murmur hash 3 algorithm, using a mock UTF-8 encoding.
+        /// The UTF-8 conversion ignores the possibilities of unicode planes other than the 0th.
+        /// That is, it simply converts char values to one, two, or three bytes according to
+        /// the following rules:
+        /// * 0x0000 to 0x007F : 0xxxxxxx
+        /// * 0x0080 to 0x07FF : 110xxxxx 10xxxxxx
+        /// * 0x0800 to 0xFFFF : 1110xxxx 10xxxxxx 10xxxxxx
+        /// NOTE: This MUST match the StringBuilder version below.
+        /// </summary>
+        public static uint MurmurHash(uint hash, ReadOnlySpan<char> span, bool toUpper = false)
```

Comment:
(nit) do you want to make this private, so other code can only see the above method?